File: Easy/Python/1437. Check If All 1's Are at Least Length K Places Away.py
class Solution(object):
    def kLengthApart(self, nums, k):
        last = -1

        for i in range(len(nums)):
            if nums[i] == 1:
                if last != -1 and i - last <= k:
                    return False
                last = i

        return True

File: Easy/JavaScript/1437. Check If All 1's Are at Least Length K Places Away.js
var kLengthApart = function(nums, k) {
    let last = -1;

    for(let i = 0; i < nums.length; i++){
        if(nums[i] == 1){
            if(last != -1 && i - last <= k)
                return false;
            last = i;
        }
    }

    return true;
};
